R/importM.COSMIC.R

```diff
@@ -0,0 +1,348 @@
+#' (as TCGA-SampleID (note that the required information for a proper TCGA
+#' mutation barcode is not available for COSMIC data, and that the sampleID
+#' for this data is specific to COSMIC data)), row names are HGNC gene
+#' symbols, and the values represent copy number.
+#'
+#' @section COSMIC CNV :
+#' The format of the COSMIC CNV data is as follows.
+#' (through sfpt, run get with this path:
+#' /files/grch38/cosmic/v80/CosmicCompleteCNA.tsv.gz)
+#'
+#' [col number] Heading
+#' [1] CNV_ID
+#' [2] Gene ID
+#' [4] Sample/Tumour ID
+#' [6] Primary Site
+#' [7-9] Site Subtype 1-3
+#' [10] Primary Histology
+#' [11-13] Histology Subtype 1-3
+#' [14] Sample Name
+#' [15] Total Copy Number
+#' [16] Minor Allele CN
+#' [17] Mutation Type
+#' [18] Study ID
+#' [19] GRCh Coordinate System
+#' [20] Genomic Coordinates of Variation
+#'
+#' @section (Replaced with fastMap()) COSMIC HGNC:
+#' The format of the COSMIC HGNC data is as follows.
+#' (through sfpt, run get with this path:
+#' /files/grch38/cosmic/v80/CosmicHGNC.tsv.gz)
+#'
+#' [col number] Heading
+#' [1] COSMIC Gene ID
+#' [2] COSMIC Gene Name
+#' [3] Entrez ID
+#' [4] HGNC ID
+#' [5] Mutated? (does gene have coding mutations? (y/n))
+#' [6] Cancer Cencus? (is gene in cancer cencus? (y/n))
+#' [7] Expert Curated? (created by team of expert curators? (y/n))
+#'
+#' @param fNameCNV The path to a TSV file with CNV data from COSMIC.
+#' #@param fNameHGNC The path to a TSV file with HGNC mapping data from COSMIC.
+#' @param outFName The path to an output file for the rCNA RDS object. Only
+#'   required if writeToFile == TRUE. Must be .rds file type.
+#' @param writeToFile Controls whether to serialize rCNA RDS object.
+#'   FALSE by default.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return Return rCNA object of COSMIC CNV data as RDS.
+#'
+#' @family importM.COSMIC, importM.TCGA, importM.GISTIC2, importM.ProjectGenie
+#'
+#'   ## @seealso \code{\link{fastMap}} fastMap() is used internally to
+#'   map COSMIC IDs to gene symbols.
+#'
+#' @examples
+#' \dontrun{
+#' fNameCNV <- "./../test/COSMIC/testCosmicCNA.tsv"
+#'
+#' importM.COSMIC(fNameCNV,
+#'                writeToFile = FALSE,
+#'                silent = TRUE,
+#'                writeLog = FALSE)
+#' }
+#' @export
+importM.COSMIC <- function(fNameCNV,
+                           #fNameHGNC,
+                           outFName,
+                           writeToFile = FALSE,
+                           silent = FALSE,
+                           writeLog = TRUE) {
+
+    # fNameCNV <- "testCosmicCNA.tsv"
+    # outFName <- "cosmicCNV.rds"
+    # fNameHGNC <- "CosmicHGNC.tsv"
+
+    # ==== PARAMETER TYPE VALIDATION ==========================================
+
+    # Check that fNameCNV is a string.
```

Comment:
Changed to use .CheckArgs() - didn't use before because code did not compile without building package; now works.